client/minimal-relay-chain-node/src/collator_overseer.rs

```diff
@@ -0,0 +1,246 @@
+			.bitfield_signing(DummySubsystem)
+			.candidate_backing(DummySubsystem)
+			.candidate_validation(DummySubsystem)
+			.pvf_checker(DummySubsystem)
+			.chain_api(DummySubsystem)
+			.collation_generation(CollationGenerationSubsystem::new(Metrics::register(registry)?))
+			.collator_protocol({
+				let side = ProtocolSide::Collator(
+					network_service.local_peer_id().clone(),
+					collator_pair,
+					collation_req_receiver,
+					Metrics::register(registry)?,
+				);
+				CollatorProtocolSubsystem::new(side)
+			})
+			.network_bridge_rx(NetworkBridgeRxSubsystem::new(
+				network_service.clone(),
+				authority_discovery_service.clone(),
+				Box::new(network_service.clone()),
+				network_bridge_metrics.clone(),
+				peer_set_protocol_names.clone(),
+			))
+			.network_bridge_tx(NetworkBridgeTxSubsystem::new(
+				network_service.clone(),
+				authority_discovery_service.clone(),
+				network_bridge_metrics,
+				req_protocol_names,
+				peer_set_protocol_names,
+			))
+			.provisioner(DummySubsystem)
+			.runtime_api(RuntimeApiSubsystem::new(
+				runtime_client.clone(),
+				Metrics::register(registry)?,
+				spawner.clone(),
+			))
+			.statement_distribution(DummySubsystem)
+			.approval_distribution(DummySubsystem)
+			.approval_voting(DummySubsystem)
+			.gossip_support(DummySubsystem)
+			.dispute_coordinator(DummySubsystem)
+			.dispute_distribution(DummySubsystem)
+			.chain_selection(DummySubsystem)
+			.leaves(Vec::from_iter(
+				leaves
+					.into_iter()
+					.map(|BlockInfo { hash, parent_hash: _, number }| (hash, number)),
+			))
+			.activation_external_listeners(Default::default())
+			.span_per_active_leaf(Default::default())
+			.active_leaves(Default::default())
+			.supports_parachains(runtime_client)
+			.known_leaves(LruCache::new(KNOWN_LEAVES_CACHE_SIZE))
+			.metrics(metrics)
+			.spawner(spawner);
+
+		builder.build_with_connector(connector).map_err(|e| e.into())
+	}
+}
+
+pub(crate) fn spawn_overseer(
+	overseer_args: CollatorOverseerGenArgs,
+	task_manager: &TaskManager,
+	relay_chain_rpc_client: Arc<BlockChainRpcClient>,
+) -> Result<polkadot_overseer::Handle, polkadot_service::Error> {
+	let overseer_gen = CollatorOverseerGen;
+	let (overseer, overseer_handle) = overseer_gen
+		.generate(OverseerConnector::default(), overseer_args)
+		.map_err(|e| {
+			tracing::error!("Failed to initialize overseer: {}", e);
+			e
+		})?;
+
+	let overseer_handle = Handle::new(overseer_handle.clone());
+	{
+		let handle = overseer_handle.clone();
+		task_manager.spawn_essential_handle().spawn_blocking(
+			"overseer",
+			None,
+			Box::pin(async move {
+				use futures::{pin_mut, FutureExt};
+
+				let forward = forward_collator_events(relay_chain_rpc_client, handle).fuse();
+
+				// let forward = forward.fuse();
+				let overseer_fut = overseer.run().fuse();
+
+				pin_mut!(overseer_fut);
+				pin_mut!(forward);
+
+				select! {
+					_ = forward => (),
+					_ = overseer_fut => (),
+					complete => (),
```

Comment:
```suggestion
```
I don't think this branch is necessary, as the select terminates as soon as one of the two completes.